.travis.yml

```diff
@@ -1,58 +1,58 @@
-  - NPM_PACKAGE_VERSION="$(npm view tplant version)"
-  - echo "NPM_PACKAGE_VERSION=$NPM_PACKAGE_VERSION"
-  - echo "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST"
-  - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH"
-  - |
-    if [ "$REPO_VERSION" != "$TPLANT_VERSION" ]; then
-      echo -e '\033[0;31mPackage version and commander version must be the same.';
-      travis_terminate 1;
-    fi
-  - |
-    if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$TRAVIS_BRANCH" == "master" ] && [ "$REPO_VERSION" == "$NPM_PACKAGE_VERSION" ]; then
-      echo -e '\033[0;31mPackage version must be higher than the current one.';
-      travis_terminate 1;
-    fi
-  - npm run lint
-
+- REPO_VERSION="$(npx -c 'echo "$npm_package_version"')"
+- echo "REPO_VERSION=$REPO_VERSION"
+- npm run build
+- TPLANT_VERSION="$(node dist/index.js -V)"
+- echo "TPLANT_VERSION=$TPLANT_VERSION"
+- NPM_PACKAGE_VERSION="$(npm view @jeff-tian/tplant version)" || NPM_PACKAGE_VERSION=0
+- echo "NPM_PACKAGE_VERSION=$NPM_PACKAGE_VERSION"
+- echo "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST"
+- echo "TRAVIS_BRANCH=$TRAVIS_BRANCH"
+- |
+  if [ "$REPO_VERSION" != "$TPLANT_VERSION" ]; then
+    echo -e '\033[0;31mPackage version and commander version must be the same.';
+    travis_terminate 1;
+  fi
+- |
+  if [ "$TRAVIS_PULL_REQUEST" != "false" ] && [ "$TRAVIS_BRANCH" == "fork" ] && [ "$REPO_VERSION" == "$NPM_PACKAGE_VERSION" ]; then
+    echo -e '\033[0;31mPackage version must be higher than the current one.';
+    travis_terminate 1;
+  fi
+- npm run lint
 jobs:
   include:
-    - stage: "Github Deployment"
-      if: branch = master AND type = push
-      node_js: "lts/*"
-      install: skip
-      before_script: skip
-      script: skip
-      before_deploy:
-        - REPO_VERSION="$(npx -c 'echo "$npm_package_version"')"
-        - echo "REPO_VERSION=$REPO_VERSION"
-        - git tag -f "$REPO_VERSION"
-        - echo "New Tag $(git describe --tags)";
-        
-      deploy:
-        provider: releases
-        api_key: $GITHUB_TOKEN
-
-    - stage: "npm Deployment"
-      if: tag IS present
-      node_js: "lts/*"
-      before_script: skip
-      script: skip
-      deploy:
-        provider: npm
-        email: "brian.folts@gmail.com"
-        api_key: $NPM_API_KEY
-        skip_cleanup: true
-        on:
-          tags: true
\ No newline at end of file
+  - stage: Github Deployment
+    node_js: lts/*
+    install: skip
+    before_script: skip
+    script: skip
+    before_deploy:
+    - REPO_VERSION="$(npx -c 'echo "$npm_package_version"')"
+    - echo "REPO_VERSION=$REPO_VERSION"
+    - git tag -f "$REPO_VERSION"
+    - echo "New Tag $(git describe --tags)";
+    deploy:
+      provider: releases
+      api_key: "$GITHUB_TOKEN"
+      on:
+        branch: fork
+  - stage: npm Deployment
+    if: tag IS present
+    node_js: lts/*
+    before_deploy:
+      - ls
+      - echo "$NPM_TOKEN"
+    deploy:
+      edge: true
+      provider: npm
+      email: jeff.tian@outlook.com
```

Comment:
This was probably configured to merge to your NPM but will conflict with the current setup. If the goal is to merge with this `bafolts` master I would change this back.

Reply:
It's mis-operation. I just need the feature yesterday so published one to @jeff-tian/tplant. Reverted it.